iOS/src/toga_iOS/widgets/activityindicator.py

```diff
@@ -0,0 +1,34 @@
+from rubicon.objc import CGSize
+
+from toga_iOS.libs import UIActivityIndicatorView, UIView
+from toga_iOS.widgets.base import Widget
+
+
+class ActivityIndicator(Widget):
+    def create(self):
+        self._activity_indicator = UIActivityIndicatorView.new()
+        self._activity_indicator.hidesWhenStopped = True
+        self._activity_indicator.translatesAutoresizingMaskIntoConstraints = False
+
+        self.native = UIView.new()
+        self.native.addSubview(self._activity_indicator)
+        self.native.translatesAutoresizingMaskIntoConstraints = False
+        self.native.sizeToFit()
```

Comment:
Is there a reason that both the UIView and UIActivatorView is needed? Is it not possible to use UIActivityIndicatorView directly as the native widget? If it isn't, there should be a docstring here describing the reason; if it is... then that's an easy cleanup :-)

Reply:
For this one I had to wrap it in a parent UIView, because for some reason the UIActivityIndicatorView shows up even when stopped, regardless of whether I set the `hidesWhenStopped` property to `true` - I'll add it in the docstrings in case someone figures out what is wrong, but for now this was the only way I could think of that was relatively clean

Reply:
@freakboy3742 I added some information about the issue in a comment here: https://github.com/beeware/toga/pull/2829/commits/7d98567661927e43d5523302a7c3d867a9b92c82

Reply:
Ok - you successfully nerd sniped me here :-)

I've worked out what is going on. `hidesWhenStopped` is implemented as a wrapper around the literal `setHidden()` implementation - so stopping the spinner isn't just making the image disappear, it's literally hiding the widget.

The catch: when Toga starts up, it applies styles to all widgets - and one of the styles it applies is visibility. So - when you don't have the wrapper, Toga creates the widget, sets `hiddenWhenStopped`, and then sets the widget to be visible... which means you can then see it.

This doesn't happen when the widget is wrapped by the UIView because the hidden attribute of the container view is separate from the hidden attribute of the spinner itself.

So - the fix here is to override `set_hidden()` on the iOS ActivityIndicator so that it takes into account whether the spinner is running before applying visibility on the underlying widget.

This *might* have an impact on the testbed test - the probe evaluating visibility might also need to be adjusted.

Reply:
Thanks for the explanation! I was too stuck at thinking that it may be an issue with the bridge calls that I didn't think to look at the base widget 😅 I'll investigate, will have a look at the potential testbed changes as well

Reply:
I'm happy for the test improvement to be a separate PR.